driver-core/src/test/unit/com/mongodb/operation/CommandOperationHelperSpecification.groovy

```diff
@@ -185,23 +185,23 @@ class CommandOperationHelperSpecification extends Specification {
         given:
         def dbName = 'db'
         def command = BsonDocument.parse('''{findAndModify: "coll", query: {a: 1}, new: false, update: {$inc: {a :1}}, txnNumber: 1}''')
-        def serverDescription = Mock(ServerDescription)
-        def connectionDescription = Mock(ConnectionDescription)
-        def commandCallback = new SingleResultCallback() {
-            @Override
-            void onResult(Object result, Throwable t) {
-                this.result = command
-                this.t = null
-            }
+        def serverDescription = Stub(ServerDescription)
+        def connectionDescription = Stub(ConnectionDescription) {
+            getMaxWireVersion() >> getMaxWireVersionForServerVersion([4, 0, 0])
+            getServerType() >> ServerType.REPLICA_SET_PRIMARY
         }
         def commandCreator = Mock(CommandOperationHelper.CommandCreatorAsync) {
-            create(serverDescription, connectionDescription, commandCallback) >> { it[0].onResult(command, null) }
+            create(serverDescription, connectionDescription, _) >> {
```

Comment:
This was the root cause of this test failure.  The actual parameters didn't match the expect ones.

Reply:
This illustrates my lack of full understanding of how a Groovy test is constructed. All tests now pass in driver-core.

Reply:
Diagnosing Mock failures is a dark art.  I can walk you through how I managed it.